api/src/main/java/jakarta/data/metamodel/StaticMetamodel.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2023,2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import jakarta.data.Sort;
+
+/**
+ * <p>Annotates a class to serve as a static metamodel for an entity,
+ * enabling type-safe access to entity attribute names and related objects,
+ * such as {@link Sort}s for an attribute.</p>
+ *
+ * <p>For example, for the following entity,</p>
+ *
+ * <pre>
+ * &#64;Entity
+ * public class Person {
+ *     &#64;Id
+ *     public long ssn;
+ *
+ *     &#64;Embedded
+ *     public Name name;
+ *
+ *     public int yearOfBirth;
+ * }
+ *
+ * &#64;Embeddable
+ * public class Name {
+ *     public String first;
+ *     public String last;
+ * }
+ * </pre>
+ *
+ * <p>You can define a static metamodel as follows,</p>
+ *
+ * <pre>
+ * &#64;StaticMetamodel(Person.class)
+ * public class _Person {
```

Comment:
I know that `Person_`is already taken from Jakarta Persistence.

Can we use another suffix instead of having a prefix?

I mean, for us, is more natural, using IDE, explore suffix at the code, for example, given a Person in a code we might have:

- PersonEntity
- PersonService
- PersonRepository
- PersonController


In general, by conversion, we work mostly with suffixes instead of prefixes. Some suggestions:

- Person_Metadata
- Person__
- PersonMetadata

Reply:
My motivation for suggesting the prefix `_` was that I figured it was actually more ergonomic in the IDE: i.e. it's nicer to type `_Bo<ctrl-space>` than `Bo<ctrl-space>_`.

If you guys really hate the prefix underscore, I would suggest something like `Entity_Person`.

Reply:
> My motivation for suggesting the prefix `_` was that I figured it was actually more ergonomic in the IDE: i.e. it's nicer to type `_Bo<ctrl-space>` than `Bo<ctrl-space>_`.
> 
> If you guys really hate the prefix underscore, I would suggest something like `Entity_Person`.

My own preference is that I liked the single underscore because it also makes the name as concise as possible, making the code easier to read and understand,

```
page = people.findByAgeBetween(20, 65, Pageable.ofSize(10).sortBy(_Person.lastName.asc(),
                                                                  _Person.firstName.asc(),
                                                                  _Person.id.asc());
```

rather than

```
page = people.findByAgeBetween(20, 65, Pageable.ofSize(10).sortBy(Person_Metadata.lastName.asc(),
                                                                  Person_Metadata.firstName.asc(),
                                                                  Person_Metadata.id.asc());
```

Reply:
🤔 I did not know that. Thank you for explaining.
If it is more ergonomic to the IDE, let's keep this way.